File: DependencyInjection/DependencyInjection/src/RobotsWithDI/Robot.java
package RobotsWithDI;

public class Robot {
    private String model;
    private String serial;
    private MessageSender messageSender;

    public Robot(String model, String serial, MessageSender messageSender){
        this.model = model;
        this.serial = serial;
        this.messageSender = messageSender;
    }

    public void sendRobotInformation(){
        String robotInfo = getRobotInformation();
        messageSender.sendInformation(robotInfo);
    }

    private String getRobotInformation(){
        String robotInfo = String.format("Model: %s\nSerial: %s", model, serial);
//        Any extra steps for gathering/formatting info
        return robotInfo;
    }

//    Other methods

}

File: DependencyInjection/DependencyInjection_Ruby/robot.rb
class Robot
    attr_accessor :model, :serial, :message_sender
    def initialize(model, serial, message_sender)
        @model = model
        @serial = serial
        @message_sender = message_sender
    end

    def send_robot_information
        robot_info = get_robot_information();
        message_sender.send_information(robot_info)
    end

    # Other important methods
    private

    def get_robot_information
        "Model: #{model} \nSerial: #{serial}"
    end
end